src/neural/cuda/layers.cc

```diff
@@ -900,6 +901,251 @@ FusedWinogradConvSELayer<DataType>::~FusedWinogradConvSELayer() {
   }
 }
 
+
+template <typename DataType>
+ResidualBlock<DataType>::ResidualBlock(
+    BaseLayer<DataType>* ip, int C, bool se, int se_k, bool use_gemm_ex, bool first, bool last)
+    : BaseLayer<DataType>(C, 8, 8, ip),
+      c_input_(C),
+      has_se_(se),
+      se_k_(se_k),
+      use_gemm_ex_(use_gemm_ex),
+      first_block_(first),
+      last_block_(last) {
+  // Allocate memory for weights (filter tensor) and biases.
+  const size_t weight_size = sizeof(DataType) * C * C * 3 * 3;
+
+  const size_t blas_size = sizeof(DataType) * C;
+  ReportCUDAErrors(cudaMalloc(&biases0_, blas_size));
+  ReportCUDAErrors(cudaMalloc(&biases1_, blas_size));
+
+  // 6x6 transformed filter size, for 3x3 convolution
+  ReportCUDAErrors(cudaMalloc(&transformed_weights0_, weight_size * 4));
+  ReportCUDAErrors(cudaMalloc(&transformed_weights1_, weight_size * 4));
+
+  if (has_se_) {
+    const size_t num_weights1 = C * se_k_;
+    const size_t num_weights2 = num_weights1 * 2;
+    const size_t num_biases1 = se_k_;
+    const size_t num_biases2 = 2 * C;
+
+    const size_t weight_size1 = sizeof(DataType) * num_weights1;
+    const size_t weight_size2 = sizeof(DataType) * num_weights2;
+    const size_t biases_size1 = sizeof(DataType) * num_biases1;
+    const size_t biases_size2 = sizeof(DataType) * num_biases2;
+
+    ReportCUDAErrors(cudaMalloc(&w1_, weight_size1));
+    ReportCUDAErrors(cudaMalloc(&w2_, weight_size2));
+    ReportCUDAErrors(cudaMalloc(&b1_, biases_size1));
+    ReportCUDAErrors(cudaMalloc(&b2_, biases_size2));
+  }
+}
+
+template <typename DataType>
+void ResidualBlock<DataType>::LoadWeights0(float* pfilter,
+                                           float* pBias,
+                                           void* scratch) {
+
+  const size_t weight_size = sizeof(float) * c_input_ * C * 3 * 3;
+  const size_t blas_size = sizeof(float) * C;
```

Comment:
This should be `bias_size` (I know it is like that in other kernels as well).

Reply:
thanks. It was a typo that got copied over to all places. Fixed.